kitty: 0.27.1 -> 0.28.1

adamcstephens opened the pull request:
###### Description of changes

https://github.com/kovidgoyal/kitty/blob/v0.28.0/docs/changelog.rst#0280-2023-04-15

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

winterqt commented on pkgs/applications/terminal-emulators/kitty/update.sh:
```diff
@@ -0,0 +1,36 @@
+fi
+
+HASH=$(nix-prefetch-url --quiet --unpack https://github.com/kovidgoyal/kitty/archive/refs/tags/${LATEST_TAG}.tar.gz)
+HASH=$(nix hash to-sri --type sha256 "$HASH")
```
```suggestion
HASH=$(nix --experimental-features=nix-command hash to-sri --type sha256 "$HASH")
```

adamcstephens replied:
FYI it's `--experimental-features nix-command`. The = is invalid apparently.

winterqt replied:
I forgot the CLI parser does not make any sense, thanks.

Luflosi commented:
Could the update script also verify the GPG signature?

adamcstephens commented:
> Could the update script also verify the GPG signature?

I've added verification of the git tag signature using the GitHub API.

amarshall reviewed (approved):
Verified builds and basic behavior works on x86_64-linux & x86_64-darwin. I haven’t reviewed the update script.

adamcstephens committed:
kitty: 0.27.1 -> 0.28.1

adamcstephens commented:
I've removed the bash update script, as @lilyinstarlight pointed out that the latest nix-update could work with this python/go package. Split go-modules into its own attribute and verified the vendor hash is now updated by nix-update.

adamcstephens commented:
@Luflosi do you have concerns about switching to nix-update instead of the bash script, given that it doesn't validate the GPG signature in any way? I'd appreciate your feedback/review on this if you have the time. Thanks!

Luflosi commented:
I'm fine with using nix-update.
Since kitty hasn't been compatible with my old macOS version for some time now, I'm using an old version of kitty and am not particularly interested in maintaining the current version. I plan to actively maintain the kitty package again when I get a new laptop (with NixOS) some time this year.

Luflosi commented on pkgs/applications/terminal-emulators/kitty/default.nix:
```diff
@@ -99,16 +100,16 @@ buildPythonApplication rec {
 
-  configurePhase = let
-    goModules = (buildGoModule {
+  go-modules =(buildGoModule {
```
```suggestion
  go-modules = (buildGoModule {
```

adamcstephens replied:
Fixed. I look forward to a formatting standard...

ocfox commented:
Result of `nixpkgs-review pr 226297` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>kitty</li>
    <li>kitty.kitten</li>
    <li>kitty.shell_integration</li>
    <li>kitty.terminfo</li>
  </ul>
</details>

ocfox reviewed (commented):
Except for format error, LGTM, and works fine on darwin-aarch64.

lilyinstarlight reviewed (approved):
Looks good to me and seems to work correctly when testing on my system. Thank you!
